packages/thirdweb/src/extensions/erc1155/drops/write/claimTo.ts

```diff
@@ -95,45 +31,45 @@ export type ClaimToParams = {
  * @returns A promise that resolves with the submitted transaction hash.
  */
 export function claimTo(options: BaseTransactionOptions<ClaimToParams>) {
-  return prepareContractCall({
+  return claim({
     contract: options.contract,
-    method: CLAIM_ABI,
-    params: async () => {
+    async asyncParams() {
       const cc = await getActiveClaimCondition({
         contract: options.contract,
         tokenId: options.tokenId,
       });
-      // TODO implement fetching merkle data
+      let proofData = {
+        currency: cc.currency,
+        proof: [] as `0x${string}`[],
+        quantityLimitPerWallet: cc.quantityLimitPerWallet,
+        pricePerToken: cc.pricePerToken,
+      };
       if (cc.merkleRoot !== padHex("0x", { size: 32 })) {
-        throw new Error("Allowlisted claims not implemented yet");
+        const claimerProof = await fetchProofsForClaimer({
+          contract: options.contract,
+          claimer: options.from || options.to, // receiver and claimer can be different, always prioritize the claimer for allowlists
+          merkleRoot: cc.merkleRoot,
+        });
+
+        if (claimerProof) {
+          proofData = claimerProof;
+        }
```

Comment:
this has code smell, I think we should just branch via return statements instead, the `let` is gonna cause problems here later